src/components/my-widgets-export.jsx

```diff
@@ -0,0 +1,202 @@
+					: undefined) > 0
+			) {
+				tmpOps = tmpOps.concat(inst.widget.meta_data.playdata_exporters)
+			}
+
+			let options = new Array(scores.length).fill(false)
+			options[0] = true
+			setSemesterOptions(options)
+			setExportOptions(tmpOps)
+		}
+
+		setExportType(tmpOps[0])
+	}, [])
+	
+	// Sets all values to checked or unchecked when checkall is clicked
+	useEffect(() => {
+		if (checkAll.noChange === true) return
+
+		if (semesterOptions.length > 0) {
+			let arr = new Array(scores.length).fill(checkAll.check)
+			setSemesterOptions(arr)
+		}
+	}, [checkAll])
+
+	// Sets selected semesters and their respective header text
+	useEffect(() => {
+		if (semesterOptions.length > 0) {
+			let str = ""
+			let str_cpy = ""
+			
+			for (let i = 0; i < semesterOptions.length; i++) {
+				if (semesterOptions[i]) {
+					if (str !== "") {
+						str += ", "
+						str_cpy += ","
+					}
+
+					str += scores[i].year + " " + scores[i].term
+					str_cpy += scores[i].year + "-" + scores[i].term
+				}
+			}
+
+			if (str === "") {
+				str = "No semester selected"
+			}
+
+			// sets check all if all options are checked/unchecked
+			if (semesterOptions.includes(true) && !semesterOptions.includes(false))
+				setCheckAll({check: true, noChange: true})
+			else {
+				setCheckAll({check: false, noChange: true})
+			}
+
+			setHeader(str)
+			setSelectedSemesters(str_cpy)
+		}
+	}, [semesterOptions])
+
+	// Used on semester selection
+	const semesterCheck = (index) => {
+		let arr = [... semesterOptions]
+		arr[index] = !arr[index]
+		setSemesterOptions(arr)
+	}
+
+	return (
+		<Modal onClose={onClose} noGutter>
+			<div className="export-modal">
+				<div className="top-bar">
+					<div className="content">
+						<span className="title">Export</span>
+						<span className="semester-btn" onClick={() => {setShowOptions(!showOptions)}}>
+							{ showOptions
+								? "Hide"
+								: "Semesters"
+							}
+						</span>
+					</div>
+					<div className="underline"></div>
+				</div>
+				<div className="semester-content">
+					<h3>{header}</h3>
+					<div className="score-table">
+						<p id="export-scores-description">
+							<span className="highlight">Export Scores{" "}</span>
+							provides a means of exporting student score information in .CSV
+							format, much like an excel spreadsheet. Teachers can use the scores
+							to analyze, compare, and gauge class performance. In addition, teachers
+							can also download a CSV containing a widget's question and answer
+							set by selecting the Questions and Answers option from the drop-down
+							menu. Download options may vary by widget, as some widgets
+							provide specialized export options.
+						</p>
+						<div className="download-controls">
+							<select value={exportType} onChange={(e) => {setExportType(e.target.value)}} >
+								{ 
+									exportOptions.map((val, index) => {
+										return (<option key={index} value={val}>{val}</option>)
+									})
```

Comment:
take advantage of implicit arrow function object returns

```suggestion
									exportOptions.map((val, index) => <option key={index} value={val}>{val}</option>)
```